### PR TITLE
Update Warning Section For FastScan

### PR DESCRIPTION
**Pull request by hongyunyan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/use-fastscan.md [hunk 1]
```diff
@@ -12,6 +12,39 @@ aliases: ['/zh/tidb/dev/sql-statement-set-tiflash-mode/','/zh/tidb/dev/dev-guide
...
++------+------+
+| a    | b    |
++------+------+
+|    1 |    2 |
+|    4 |    2 |
+|   10 |   20 |
++------+------+
+```
+
+> 仅表中当所有的数据都进行了 compact 整理，并且旧版本数据超过 GC 时间后，相同主键的多版本数据才会被删除，此时 FastScan 的查询结果和非 FastScan 模式下保持相同。数据 compact 的时机受多种因素的自动触发，用户也可以通过 [`Alter Table ... Compact`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据 compact 整理。
```
Comment (flowbehappy) [inline tiflash/use-fastscan.md:46]:
```suggestion
> 仅表中当所有的数据都进行了 compact 整理，并且旧版本数据超过 GC 时间后，相同主键的多版本数据才会被删除。数据 compact 的时机受多种因素的自动触发，用户也可以通过 [`Alter Table ... Compact`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据 compact 整理。 
```

最好不要提“保持相同” 吧，容易引起误解

File: tiflash/use-fastscan.md [hunk 2]
```diff
@@ -12,6 +12,39 @@ aliases: ['/zh/tidb/dev/sql-statement-set-tiflash-mode/','/zh/tidb/dev/dev-guide
...
+> **警告：**
+> 当开启 FastScan 功能时，查询结果会包含相同主键的多个版本的数据, 如下所示：
+
+> 
+```
+create table t1 (a int primary key, b int);
+alter table t1 set tiflash replica 1;
+insert into t1 values(1,2);
+insert into t1 values(10,20);
+update t1 set a = 4 where b = 2;
```
Comment (flowbehappy) [inline tiflash/use-fastscan.md:24]: 建议更新 b 比较直观。

File: tiflash/use-fastscan.md [hunk 3]
```diff
@@ -12,6 +12,39 @@ aliases: ['/zh/tidb/dev/sql-statement-set-tiflash-mode/','/zh/tidb/dev/dev-guide
...
++------+------+
+| a    | b    |
++------+------+
+|    1 |    2 |
+|    4 |    2 |
+|   10 |   20 |
++------+------+
+```
+
+> 仅表中当所有的数据都进行了 compact 整理，并且旧版本数据超过 GC 时间后，相同主键的多版本数据才会被删除。数据 compact 的时机受多种因素的自动触发，用户也可以通过 [`Alter Table ... Compact`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据 compact 整理。
```
Comment (flowbehappy) [inline tiflash/use-fastscan.md:46]:
```suggestion
> 在这些旧数据（相同主键的多版本旧数据和已经被 delete 语句删除的数据）被进行了 compact 整理，并且它们的数据版本小于 GC safe point 之后，才会被真正物理清理。此后在 FastScan 模式下不再返回这些数据 。数据被 compact 的时机受多种因素的自动触发，用户也可以通过 [`Alter Table ... Compact`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据 compact 整理。 
```

File: tiflash/use-fastscan.md [hunk 4]
```diff
@@ -66,8 +99,8 @@ TiFlash 存储层的数据主要存放在 Delta 层和 Stable 层。
 在默认状态下（即未开启 FastScan 功能），TableScan 算子过程整体包括了以下步骤：
 
 1. Read data：在 Delta 层和 Stable 层分别建立数据流，进行各自数据的读取。
-2. Sort Merge：将步骤 1 中建立的数据流进行合并，并且将数据按照 (handle, version) 顺序排列返回。
+2. Sort Merge：将步骤 1 中建立的数据流进行合并，并且将数据按照 (主键列, 时间戳列) 顺序排列返回。
 3. Range Filter：根据读取范围限制，对步骤 2 中的数据进行过滤筛选并返回。
-4. MVCC + Column Filter：对步骤 3 中的数据进行 MVCC 过滤，同时过滤掉不需要的列并返回。
+4. MVCC + Column Filter：对步骤 3 中的数据进行 MVCC 过滤（即依照查询需要的数据版本，根据主键列以及时间戳列进行过滤掉其他版本的数据行），同时进行 Column 过滤（即过滤掉查询请求中不需要的列）并返回。
```
Comment (flowbehappy) [inline tiflash/use-fastscan.md:104]:
```suggestion
4. MVCC + Column Filter：对步骤 3 中的数据进行 MVCC 过滤（即依照查询需要的数据版本，根据主键列以及时间戳列筛选出正确的数据版本），同时进行 Column 过滤（即过滤掉查询请求中不需要的列）并返回。
```

File: tiflash/use-fastscan.md [hunk 5]
```diff
@@ -12,6 +12,39 @@ aliases: ['/zh/tidb/dev/sql-statement-set-tiflash-mode/','/zh/tidb/dev/dev-guide
...
++------+------+
+| a    | b    |
++------+------+
+|    1 |    2 |
+|    1 |    4 |
+|   10 |   20 |
++------+------+
+```
+
+> 这些旧数据（相同主键的多版本旧数据和已经被 delete 语句删除的数据）在被进行了 compact 整理，并且它们的数据版本小于 GC safe point 之后，才会被真正物理清理。此后在 FastScan 模式下不再返回这些数据。数据被 compact 的时机受多种因素的自动触发，用户也可以通过 [`Alter Table ... Compact`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据 compact 整理。
```
Comment (flowbehappy) [inline tiflash/use-fastscan.md:46]:
```suggestion
> 这些旧数据（相同主键的多版本旧数据和已经被删除的数据）在被进行了 compact 整理，并且它们的数据版本小于 GC safe point 之后，才会被真正物理清理。此后在 FastScan 模式下不再返回这些数据。数据被 compact 的时机受多种因素的自动触发，用户也可以通过 [`Alter Table ... Compact`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据 compact 整理。 
```

**Review by flowbehappy — APPROVED** [review]
LGTM

File: tiflash/use-fastscan.md [hunk 6]
```diff
@@ -12,6 +12,39 @@ aliases: ['/zh/tidb/dev/sql-statement-set-tiflash-mode/','/zh/tidb/dev/dev-guide
 
 某些 OLAP 对查询结果精度可以容忍一定误差。如果对查询性能有更高要求，可以在 session 级别或 global 级别开启 FastScan 功能，你可以通过修改变量 `tiflash_fastscan` 的值来选择是否启用 FastScan 功能。
 
+> **警告：**
+> 当开启 FastScan 功能时，查询结果会包含相同主键的多个版本的数据，或者已经被删除的数据。如下所示：
+
+> 
+```
```
Comment (qiancai) [inline tiflash/use-fastscan.md:19]:
```suggestion
```sql
```

File: tiflash/use-fastscan.md [hunk 7]
```diff
@@ -12,6 +12,39 @@ aliases: ['/zh/tidb/dev/sql-statement-set-tiflash-mode/','/zh/tidb/dev/dev-guide
...
++------+------+
+| a    | b    |
++------+------+
+|    1 |    2 |
+|    1 |    4 |
+|   10 |   20 |
++------+------+
+```
+
+> 这些旧数据（相同主键的多版本旧数据和已经被删除的数据）在被进行了 compact 整理，并且它们的数据版本小于 GC safe point 之后，才会被真正物理清理。此后在 FastScan 模式下不再返回这些数据。数据被 compact 的时机受多种因素的自动触发，用户也可以通过 [`Alter Table ... Compact`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据 compact 整理。
```
Comment (qiancai) [inline tiflash/use-fastscan.md:46]:
```suggestion
对于这些旧数据，TiDB 存储节点在后台会自动发起数据整理（Compaction）。当这些旧数据已经被后台整理且它们的数据版本小于 GC safe point 之后，才会被物理清理。此后在 FastScan 模式下将不再返回这些数据。旧数据被整理的时机受多种因素的自动触发，你也可以通过 [`ALTER TABLE ... COMPACT`](/sql-statements/sql-statement-alter-table-compact.md) 手动触发数据整理。 
```

File: tiflash/use-fastscan.md [hunk 8]
```diff
@@ -66,8 +99,8 @@ TiFlash 存储层的数据主要存放在 Delta 层和 Stable 层。
 在默认状态下（即未开启 FastScan 功能），TableScan 算子过程整体包括了以下步骤：
 
 1. Read data：在 Delta 层和 Stable 层分别建立数据流，进行各自数据的读取。
-2. Sort Merge：将步骤 1 中建立的数据流进行合并，并且将数据按照 (handle, version) 顺序排列返回。
+2. Sort Merge：将步骤 1 中建立的数据流进行合并，并且将数据按照 (主键列, 时间戳列) 顺序排列返回。
```
Comment (qiancai) [inline tiflash/use-fastscan.md:102]:
```suggestion
2. Sort Merge：将步骤 1 中建立的数据流进行合并，并且将数据按照 (主键列，时间戳列) 顺序排列返回。
```

File: tiflash/use-fastscan.md (hunk 7 above)
Comment (hongyunyan) [reply tiflash/use-fastscan.md:46]: TiDB 存储节点  是不是可以改为 TiFlash 这样会更直接一些呢，不然感觉会有 TiKV 和 TiFlash 的歧义

Comment (qiancai) [reply tiflash/use-fastscan.md:46]: 这是参考了 https://docs.pingcap.com/zh/tidb/dev/sql-statement-alter-table-compact 这里，写的是 “TiDB 存储节点在后台会自动发起数据整理（Compaction）“。 那要不把 sql-statement-alter-table-compact.md 这里的 “TiDB 存储节点” 也改成 TiFlash，保持两边一致

Comment (hongyunyan) [reply tiflash/use-fastscan.md:46]: 我觉得 https://docs.pingcap.com/zh/tidb/dev/sql-statement-alter-table-compact  这里的 “TiDB 存储节点在后台会自动发起数据整理（Compaction）“ 他想表达的是 tikv 和 tiflash 内部都有自动 compact 的机制，感觉是个总述的意思。但是我们这边其实想特指 tiflash 本身的 compact 时机。

Comment (qiancai) [reply tiflash/use-fastscan.md:46]: 好的，我修改成 “对于这些旧数据，TiFlash 在后台会自动发起数据整理（Compaction）” 了

**Review by qiancai — APPROVED** [review]
/lgtm

**Review by ran-huang — APPROVED** [review]

**Pull request merged**